bundles/ch.elexis.core.ui.documents/src/ch/elexis/core/ui/documents/views/DocumentsMetaDataDialog.java

```diff
@@ -218,60 +226,97 @@ public void widgetSelected(SelectionEvent ev){
+				tDocumentClass.setText(proposal.getLabel());
+				tDocumentClass.setData(PROPOSAL_RET_OBJ, ((CodingContentProposal) proposal).getCoding());
+				tDocumentClass.setSelection(tDocumentClass.getText().length());
+			}
+		});
+		ICoding docClassCode = documentReference.getDocumentClass();
+		if (docClassCode != null) {
+			tDocumentClass.setText(docClassCode.getDisplay());
+			tDocumentClass.setData(PROPOSAL_RET_OBJ, docClassCode);
+		}
+		
+
+		new Label(ret, SWT.NONE).setText(Messages.DocumentsView_PracticeSetting);
+		tPracticeSetting = SWTHelper.createText(ret, 1, SWT.NONE);
+		ContentProposalAdapter tPracticeSettingContentProposal =
+			new ContentProposalAdapter(tPracticeSetting, new TextContentAdapter(),
+				new PracticeSettingProposalProvider(), null, null);
+		tPracticeSettingContentProposal.addContentProposalListener(new IContentProposalListener() {
+			@Override
+			public void proposalAccepted(IContentProposal proposal){
+				tPracticeSetting.setText(proposal.getLabel());
+				tPracticeSetting.setData(PROPOSAL_RET_OBJ, ((CodingContentProposal) proposal).getCoding());
+				tPracticeSetting.setSelection(tPracticeSetting.getText().length());
+			}
+		});
+		ICoding pracSetttingCode = documentReference.getPracticeSetting();
+		if (pracSetttingCode != null) {
+			tPracticeSetting.setText(pracSetttingCode.getDisplay());
+			tPracticeSetting.setData(PROPOSAL_RET_OBJ, pracSetttingCode);
 		}
 	}
 	
-	public Optional<IDocumentReference> getDocumentReference(IDocument document){
-		if (FindingsServiceHolder.getiFindingsService() != null && document != null
-			&& document.getId() != null) {
-			List<IDocumentReference> documentReferences =
-				FindingsServiceHolder.getiFindingsService().getDocumentFindings(document.getId(),
-					IDocumentReference.class);
-			if (documentReferences.size() > 1) {
-				LoggerFactory.getLogger(getClass())
-					.warn("Got more than one DocumentReferences for document id ["
-						+ document.getId() + "] using first");
-			}
-			return documentReferences.stream().findFirst();
+	private IDocumentReference getOrCreateDocumentReference(IDocument document){
+		List<IDocumentReference> documentReferences = FindingsServiceHolder.getiFindingsService()
+			.getDocumentFindings(document.getId(), IDocumentReference.class);
+		if (documentReferences.size() > 1) {
+			LoggerFactory.getLogger(getClass())
+				.warn("Got more than one DocumentReferences for document id [" + document.getId()
+					+ "] using first");
+		}
+		if (!documentReferences.isEmpty()) {
+			return documentReferences.get(0);
 		}
-		return Optional.empty();
+		
+		// no document reference found - create new entry with mandatory fields
+		IDocumentReference documentReference =
+			FindingsServiceHolder.getiFindingsService().create(IDocumentReference.class);
+		documentReference.setPatientId(document.getPatient().getId());
+		documentReference.setDocument(document);
+		FindingsServiceHolder.getiFindingsService().saveFinding(documentReference);
```

Comment:
Hier speichern wird vmtl. ein Problem wenn ein Dokument importiert wird, und der Dialog abgebrochen wird.

Reply:
Nein, da hier schon ein Document in der DB gespeichert ist.
Hintergrund:
Dzt wird nach dem Import unabhängig ob Dlg gespeichert wird oder abgebrochen, eine Document angelegt. Daher wird auch ein DocRef mitangelegt auch im Ticket steht als Kommentar "jedes Bearbeiten / Dialog öffnen führt dazu das ein CH_ELEXIS_CORE_FINDINGS_DOCUMENTREFERENCE angelegt wird wenn noch kein Eintrag vorhanden ist.".

Soll ich dieses Neu-Anlage-Verhalten generel für Dokumente & DocRef ändern ? Oder sollen wir das in eimem seperaten Ticket behandeln ?

Reply:
Bitte das Verhalten für Dokumente & DocRef so ändern das im dialog mit transienten Objekten gearbeitet wird, und nur bei OK persistiert wird.